### PR TITLE
perf: (QTM-444) preload on font loading

### PR DESCRIPTION
**Pull request by douglasjr97**
## Description
https://jirasoftware.catho.com.br/browse/QTM-444

## Review guide
- [ ] Unit tests (yarn test:components)
- [ ] Regression
- [ ] Code review
- [ ] Documentation

### Browsers review
- [ ] Layout review
  - [ ] Edge
  - [ ] Firefox
  - [ ] Chrome
  - [ ] Safari
  - [ ] Mobile

### CODE REVIEW

**Review by MarcosViniciusPC — CHANGES_REQUESTED** [review]
[] Code review
[] Regression

I assume that the preload in this case caused the google font not to be applied to the quantum typography and this changed all the project screenshots. Try to understand more about preload behavior in mozilla documents.

**Review by MarcosViniciusPC — COMMENTED** [review]
And again, try to understand how the resources you use work. Copy and paste, but seek to understand, reflect. One way to do this is by consulting official documents and the main references on the subject.

**Comment by douglasjr97** [comment]
> And again, try to understand how the resources you use work. Copy and paste, but seek to understand, reflect. One way to do this is by consulting official documents and the main references on the subject.

I trusted these sources
https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel/preload
https://pt.stackoverflow.com/questions/269556/o-que-%C3%A9-rel-preload

File: components/GlobalStyle/sub-components/Typography.jsx [hunk 1]
```diff
@@ -90,9 +90,17 @@ ul, ol {
...
+      crossOrigin="anonymous"
+      rel="preload"
+      href="https://fonts.googleapis.com/css?family=Nunito+Sans:400,600,700,400i,600i,700i&display=swap"
+      as="font"
+      type="font/woff2"
+    />
     <link
       href="https://fonts.googleapis.com/css?family=Nunito+Sans:400,600,700,400i,600i,700i&display=swap"
       rel="stylesheet"
+      as="style"
```
Comment (MarcosViniciusPC) [inline components/GlobalStyle/sub-components/Typography.jsx:103]: https://developer.mozilla.org/pt-BR/docs/Web/HTML/Element/link#attr-as
I want you to read the documentation calmly, as I instructed you. Always look for content from great references on the internet. There are many people on the internet who use some resources erroneously.

File: components/GlobalStyle/sub-components/Typography.jsx [hunk 2]
```diff
@@ -90,9 +90,17 @@ ul, ol {
 const Typography = () => (
   <>
     <Style />
+    <link
+      crossOrigin="anonymous"
+      rel="preload"
+      href="https://fonts.googleapis.com/css?family=Nunito+Sans:400,600,700,400i,600i,700i&display=swap"
+      as="font"
+      type="font/woff2"
```
Comment (MarcosViniciusPC) [inline components/GlobalStyle/sub-components/Typography.jsx:98]: You point out that the style is a font, but the href points to a css. This is divergent. I believe you are relying too much on the referral links you sent in slack. Note that in the mentioned links the problem they want to solve is importing a font and not importing a css

File: components/GlobalStyle/sub-components/Typography.jsx [hunk 3]
```diff
@@ -90,9 +90,17 @@ ul, ol {
 const Typography = () => (
   <>
     <Style />
+    <link
+      crossOrigin="anonymous"
+      rel="preload"
+      href="https://fonts.googleapis.com/css?family=Nunito+Sans:400,600,700,400i,600i,700i&display=swap"
+      as="font"
```
Comment (MarcosViniciusPC) [inline components/GlobalStyle/sub-components/Typography.jsx:97]: Divergence with the href

**Review by MarcosViniciusPC — CHANGES_REQUESTED** [review]
[] Code review

File: components/GlobalStyle/sub-components/Typography.jsx (hunk 2 above)
Comment (douglasjr97) [reply components/GlobalStyle/sub-components/Typography.jsx:98]:
```suggestion
      type="font/woff2"
```
That's why I added this format, with its presence the font extension is inside the css file, understand?

File: components/GlobalStyle/sub-components/Typography.jsx (hunk 1 above)
Comment (douglasjr97) [reply components/GlobalStyle/sub-components/Typography.jsx:103]: I checked in the documentation that the "from" attribute is only used on links that use (preload).

File: components/GlobalStyle/sub-components/Typography.jsx (hunk 3 above)
Comment (douglasjr97) [reply components/GlobalStyle/sub-components/Typography.jsx:97]: The Font Attribute already references the CSS/@font-face type, which is used in this case for the google font.
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link#attr-type

image

File: components/GlobalStyle/sub-components/Typography.jsx (hunk 2 above)
Comment (douglasjr97) [reply components/GlobalStyle/sub-components/Typography.jsx:98]: I evaluated that when we use "as=Font" there is no need to use a typing

File: components/GlobalStyle/sub-components/Typography.jsx (hunk 3 above)
Comment (MarcosViniciusPC) [reply components/GlobalStyle/sub-components/Typography.jsx:102]: Very good!

**Review by MarcosViniciusPC — APPROVED** [review]
[x] Code review

**Review by rapahaeru — APPROVED** [review]

**Pull request merged**